tools/reencode.sh

```diff
@@ -0,0 +1,11 @@
+#!/bin/bash
+
+set -e
+
+srcfile="${@: -1}"
+
+tempfile=`mktemp --tmpdir oot_XXXXXXXX.c`
```

Comment:
Does this litter /tmp with files? It should probably remove the tempfile with a `trap` on exit

Reply:
the tempfile should be removed on compile errors too IMO. Something like `trap "rm -f $tempfile" EXIT` instead of only removing it if we reach the end of the script